zebra-chain/src/orchard/tests/preallocate.rs

```diff
@@ -14,82 +11,77 @@ use proptest::{prelude::*, proptest};
-        prop_assert!(serialized.len() as u64 == ACTION_SIZE)
+    fn authorized_action_size_is_small_enough(authorized_action in <AuthorizedAction>::arbitrary_with(())) {
+        let (action, spend_auth_sig) = authorized_action.into_parts();
+        let mut serialized_len = action.zcash_serialize_to_vec().expect("Serialization to vec must succeed").len();
+        serialized_len += spend_auth_sig.zcash_serialize_to_vec().expect("Serialization to vec must succeed").len();
+        prop_assert!(serialized_len as u64 == AUTHORIZED_ACTION_SIZE)
     }
-    /// Confirm that each Signature<SpendAuth> takes exactly SPEND_AUTH_SIG_SIZE bytes when serialized.
-    /// This verifies that our calculated `TrustedPreallocate::max_allocation()` is indeed an upper bound.
-    #[test]
-    fn spend_auth_sig_size_is_correct(sig in <Signature<SpendAuth>>::arbitrary_with(())) {
-        let serialized = sig.zcash_serialize_to_vec().expect("Serialization to vec must succeed");
-        prop_assert!(serialized.len() as u64 == SPEND_AUTH_SIG_SIZE)
-    }
-}
 
-proptest! {
-    #![proptest_config(ProptestConfig::with_cases(128))]
-
-    /// Verify that...
-    /// 1. The smallest disallowed vector of `Action`s is too large to fit in a Zcash block
-    /// 2. The largest allowed vector is small enough to fit in a legal Zcash block
+    /// Verify trusted preallocation for `AuthorizedAction` and its split fields
     #[test]
-    fn action_max_allocation_is_correct(action in <Action>::arbitrary_with(())) {
+    fn authorized_action_max_allocation_is_big_enough(authorized_action in <AuthorizedAction>::arbitrary_with(())) {
+        let (action, spend_auth_sig) = authorized_action.into_parts();
 
-        let max_allocation: usize = <Action>::max_allocation().try_into().unwrap();
-        let mut smallest_disallowed_vec = Vec::with_capacity(max_allocation + 1);
-        for _ in 0..(<Action>::max_allocation()+1) {
-            smallest_disallowed_vec.push(action.clone());
-        }
-        let smallest_disallowed_serialized = smallest_disallowed_vec.zcash_serialize_to_vec().expect("Serialization to vec must succeed");
-        // Check that our smallest_disallowed_vec is only one item larger than the limit
-        prop_assert!(((smallest_disallowed_vec.len() - 1) as u64) == <Action>::max_allocation());
-        // Check that our smallest_disallowed_vec is too big to be included in a valid block
-        // Note that a serialized block always includes at least one byte for the number of transactions,
-        // so any serialized Vec<Action> at least MAX_BLOCK_BYTES long is too large to fit in a block.
-        prop_assert!((smallest_disallowed_serialized.len() as u64) >= MAX_BLOCK_BYTES);
+        let (
+            smallest_disallowed_vec_len,
+            _smallest_disallowed_serialized_len,
+            largest_allowed_vec_len,
+            largest_allowed_serialized_len,
+        ) = max_allocation_is_big_enough(action);
 
```

Comment:
Security: It looks like you forgot to copy some of the tests from the example?

This test is the most important test, because it makes sure the largest allowed length is actually correct:
https://github.com/ZcashFoundation/zebra/blob/2ea1d8624d1d72abdf582953f4f1133efe9eea4f/zebra-chain/src/sapling/tests/preallocate.rs#L86-L88